cmd/satellite/gracefulexit.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/gracefulexit"
+	"storj.io/storj/satellite/satellitedb"
+)
+
+// generateGracefulExitCSV creates a report with graceful exit data for exiting or exited nodes in a given period
+func generateGracefulExitCSV(ctx context.Context, completed bool, start time.Time, end time.Time, output io.Writer) error {
+	db, err := satellitedb.New(zap.L().Named("db"), gracefulExitCfg.Database)
+	if err != nil {
+		return errs.New("error connecting to master database on satellite: %+v", err)
+	}
+	defer func() {
+		err = errs.Combine(err, db.Close())
+	}()
+
+	var nodeIDs storj.NodeIDList
+	if completed {
+		nodeIDs, err = db.OverlayCache().GetGracefulExitCompletedByTimeFrame(ctx, start, end)
+		if err != nil {
+			return err
+		}
+	} else {
+		nodeIDs, err = db.OverlayCache().GetGracefulExitIncompleteByTimeFrame(ctx, start, end)
+		if err != nil {
+			return err
+		}
+	}
+
+	w := csv.NewWriter(output)
+	var headers []string
+	if completed {
+		headers = []string{
+			"nodeID",
+			"walletAddress",
+			"nodeCreationDate",
+			"initiatedGracefulExit",
+			"completedGracefulExit",
+			"transferredGB",
+		}
+	} else {
+		headers = []string{
+			"nodeID",
+			"walletAddress",
+			"nodeCreationDate",
+			"initiatedGracefulExit",
```

Comment:
I'd suggest using the same fields in both cases, so that the same tools can consume the input both ways. Can just leave completedGracefulExit empty for this case.